LibOS/shim/src/bookkeep/shim_handle.c

```diff
@@ -97,11 +97,19 @@ static inline int init_exec_handle (struct shim_thread * thread)
                       &exec->dentry);
         set_handle_fs(exec, fs);
         if (exec->dentry) {
+            int ret = fs->d_ops->open(exec, exec->dentry, O_WRONLY);
```

Comment:
I think a comment explaining why this open happens here might be helpful (i.e., why this is the right place for this to happen).  It might also be good to have some top-level comment on the function explaining how it fits into the exec/start-up process, and what it is expected to do.